Funcionalidad filtrado por géneros grupo 6

ismagt opened the pull request:
Referente a los conflictos pull #76

ismagt committed:
arreglando algunas cuestiones de conexion y añadido menu desplegable
para elegir el genero (a falta de tokenizar)

ismagt committed:
tokenizado cada genero por separado

ismagt committed:
Añdiendo ficheros DAO para la implementacion del filtrado de peliculas
por genero, no esta funcional simplemente subida de base

ismagt committed:
Implementada funcionalidad de filtrado, a falta de limpieza y organizar
el código

ismagt committed:
incluyendo comentarios del codigo y relaciones necesarias

ismagt committed:
modificando un poco el front

ismagt committed:
arreglando error en el comentarios

dcabezas96 committed:
cambio menor corregido

ismagt committed:
incluyendo cosas front

ismagt committed:
Merge branch 'devel2' of https://github.com/Noozh/ProyectoFinal_ISI.git into devel2

dcabezas96 committed:
corregido connect con postgresql

dcabezas96 committed:
corregidos errores

dcabezas96 committed:
corregidos errores

ismagt committed:
arreglando rating, como double precision

ismagt committed:
eliminando selectAll, no se va usar

ismagt committed:
Arreglando organización en los path

ismagt committed:
Merge branch 'devel2' of https://github.com/Noozh/ProyectoFinal_ISI.git into devel2

ismagt committed:
eliminando relacionesgeneroDao, implementado con relaciones

ismagt committed:
cambiando el path de generos

ismagt committed:
Arreglando errores de importacion y fallos leves

ismagt committed:
comentando connection y quitando dependencias del pom

ismagt committed:
implementando funcionalidad múltiple, con esto se puede realizar el
filtrado para varios generos

ismagt committed:
arreglando un poco el front, para poder hacer seleccion múltiple hay que
mantener la tecla Ctrl + seleccion que desea

ismagt committed:
añadiendo files txt sin lineas repetidas

ismagt committed:
Merge branch 'devel-genero' of https://github.com/Noozh/ProyectoFinal_ISI into devel-genero

dcabezas96 committed:
arreglado pequeño bug calificación en vez de duración

ismagt committed:
Merge branch 'devel-genero' into master-genero

ismagt committed:
arreglando uri de relaciones

ismagt committed:
modificando el selectAll para que sea integro al diseño

DanielRiveraPaniagua commented on src/main/java/urjc/isi/controladores/PeliculasController.java:
```diff
@@ -10,20 +10,23 @@
 	 */
 	public PeliculasController() {
 		ps = new PeliculasService();
+		gs = new GenerosService();
```
Esto debería irse al método concreto en el que se usa, en vez de estar aquí

ismagt replied:
arreglado commit 73431b020ba525ecaaf4973563029db460a047c9

ismagt replied:
no sobra ya que se llama a searchByGenero, aprox linea 195 :
List<Generos> output = gs.getAllGeneros();

ismagt replied:
vale ya esta, finalmente lo he introducido en generos Controller

DanielRiveraPaniagua commented on src/main/java/urjc/isi/dao/implementaciones/PeliculasDAOImpl.java:
```diff
@@ -255,4 +255,20 @@ public String selectCalificacionForPelicula(String name){
+	  }
+	  return filmList;
+	}
 }
```
Esto es lo que os comenté, si lo unís como parte de selectAll mejor así también podemos aplicar un pequeño filtrado por género allí

DanielRiveraPaniagua commented on src/main/java/urjc/isi/dao/interfaces/GenericDAO.java:
```diff
@@ -3,6 +3,9 @@
 
 import java.util.List;
+
+import urjc.isi.entidades.Peliculas;
```
No debería estar aquí

dcabezas96 replied:
Arreglado en el commit número 9f9c2abedd0aaf60ddfa00b3a1ec8d3a346fe489

DanielRiveraPaniagua commented on src/main/java/urjc/isi/dao/interfaces/RelacionesDAO.java:
```diff
@@ -4,6 +4,12 @@
+
+import urjc.isi.entidades.Relaciones;
+
+/*import urjc.isi.entidades.Relaciones;*/
```
Innecesario

dcabezas96 replied:
Arreglado en el commit número f8bfce532b74822b7605f2d3d61c4cbc96de3284

DanielRiveraPaniagua commented on src/main/java/urjc/isi/controladores/PeliculasController.java:
```diff
@@ -183,6 +186,63 @@ public static String calificacion(Request request, Response response) throws SQL
+		    {
+		        generos =  generos + " OR " + "pg.genero='" + t[1] + "'"; 
+		    }
+		}
```
El controller no puede trabajar las querys, para eso usamos las clases dao, para aislar en ellas las bases de datos

ismagt replied:
solucionado, ya esta en peliculasDaoImpl

DanielRiveraPaniagua reviewed (changes requested):
Me gustaría ver una url con la app funcionando para ver como es. En principio éstas cosillas. Puesto que podéis filtrar por varios géneros, eso sí podría ser bueno estudiar como incluirlo en un selectAll para poder utilizarlo también.

Podéis así simplificar muchísimo la búsqueda por genéros en la query, where genero=terror or/and genero=x and genero=y ... (como lo tenéis más o menos), esto metido en un bucle dentro del selectall nos posibilita facilitar vuestro buscar a través de varios géneros. 

Si no me explico preguntadme por privado.

ismagt committed:
arreglado esa linea sobraba del import

dcabezas96 committed:
corregido error en GenericDAO

dcabezas96 committed:
Merge branch 'master-genero' of https://github.com/Noozh/ProyectoFinal_ISI into master-genero

dcabezas96 committed:
corregido error en RelacionesDAO

ismagt committed:
arreglando organizacion ya que peliculas controller no puede trabajar las querys, implementado en dao

ismagt committed:
Moviendo el searchBygenero al generos, ya que se llama generos service

ismagt committed:
añadiendo implmentacion selectAll a falta de despliegue

ismagt committed:
arreglando query

ismagt committed:
implmentando funcionalidad para que las querys esten en selectAll

ismagt committed:
eliminando linea front html

DanielRiveraPaniagua commented on src/main/java/urjc/isi/controladores/PeliculasController.java:
```diff
@@ -62,7 +63,7 @@ public static String selectAllPeliculas(Request request, Response response) thro
 		String result = "";
 		Dictionary<String,String> filter = new Hashtable<String,String>();
-		
+		System.out.println("muestra");
```
Quita esta muestra XD

dcabezas96 replied:
arreglado en el commit 96c69ce2032b80ab4b338f3165f7480fe6dec4ec

DanielRiveraPaniagua reviewed (approved):
Se tiene que aprobar a pesar de estar a medio hacer algunas partes de la funcionalidad para poder seguir progresando con géneros. Todo lo que involucra esas tablas se encuentra bien acabado.
